src/coreclr/jit/scopeinfo.cpp

```diff
@@ -403,26 +403,12 @@ void CodeGenInterface::siVarLoc::siFillRegisterVarLoc(
             break;
 #endif // !TARGET_64BIT
 
-#ifdef TARGET_64BIT
         case TYP_FLOAT:
         case TYP_DOUBLE:
-            // TODO-AMD64-Bug: ndp\clr\src\inc\corinfo.h has a definition of RegNum that only goes up to R15,
-            // so no XMM registers can get debug information.
             this->vlType       = VLT_REG_FP;
             this->vlReg.vlrReg = varDsc->GetRegNum();
             break;
 
-#else // !TARGET_64BIT
-
-        case TYP_FLOAT:
-        case TYP_DOUBLE:
-            if (isFloatRegType(type))
-            {
-                this->vlType         = VLT_FPSTK;
-                this->vlFPstk.vlfReg = varDsc->GetRegNum();
-            }
-            break;
```

Comment:
Just to confirm, this isn't necessary for ABI purposes right? Floating-point args still get returned in `ST(0)` for Windows (and I think Unix as well).

Reply:
I believe so, since this handling is around locals.